coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -43,9 +44,17 @@ def extract(content, language, docstyle):
                                    DocumentationComment found in the content.
         """
         docstyle_definition = DocstyleDefinition.load(language, docstyle)
-        return extract_documentation_with_markers(
+        doc_comment_iterator = extract_documentation_with_markers(
                         content, docstyle_definition)
 
+        doc_comment_iterator = self.instantiate_padding(
+            content, doc_comment_iterator)
+
+        doc_comment_iterator = self.instantiate_kind(
+            content, doc_comment_iterator)
```

Comment:
> This can be done much more efficiently by turning this method into a generator and changing instantiate_padding and instantiate_kind to process only one DocComment at a time.
This way we only need to iterate once over the docstrings.

Yes even I'm a bit worried about this... Although Im not good with generators and it's always been a pain in my head...But if it solves it why not we do it...Just we have to note that finally we need an updated DocComment iterator/generator which will have instantiated type and paddings.

>Actually, why do we need those three separate functions to parse the Docstring? Can't we do all the work in one go in extract_documentation_with_markers?

No, we can't as we need `range` , an instance of DocComment and whole file (`content`), to instantiate type and padding.